etherman/etherman.go

```diff
@@ -292,44 +299,54 @@ func (etherMan *Client) updateGlobalExitRootEvent(ctx context.Context, vLog type
 }
 
 // WaitTxToBeMined waits for an L1 tx to be mined. It will return error if the tx is reverted or timeout is exceeded
-func (etherMan *Client) WaitTxToBeMined(ctx context.Context, tx *types.Transaction, timeout time.Duration) error {
-	return operations.WaitTxToBeMined(ctx, etherMan.EtherClient, tx, timeout)
+func (etherMan *Client) WaitTxToBeMined(ctx context.Context, tx *types.Transaction, timeout time.Duration) (bool, error) {
+	err := operations.WaitTxToBeMined(ctx, etherMan.EthClient, tx, timeout)
+	if errors.Is(err, context.DeadlineExceeded) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return true, nil
 }
 
 // EstimateGasSequenceBatches estimates gas for sending batches
-func (etherMan *Client) EstimateGasSequenceBatches(sequences []ethmanTypes.Sequence) (*types.Transaction, error) {
-	if etherMan.IsReadOnly() {
-		return nil, ErrIsReadOnlyMode
+func (etherMan *Client) EstimateGasSequenceBatches(sender common.Address, sequences []ethmanTypes.Sequence) (*types.Transaction, error) {
+	opts, err := etherMan.getAuthByAddress(sender)
+	if err == ErrNotFound {
+		return nil, ErrPrivateKeyNotFound
 	}
-	noSendOpts := *etherMan.auth
-	noSendOpts.NoSend = true
-	tx, err := etherMan.sequenceBatches(&noSendOpts, sequences)
+	opts.NoSend = true
+
+	tx, err := etherMan.sequenceBatches(opts, sequences)
 	if err != nil {
 		return nil, err
 	}
 
 	return tx, nil
 }
 
-// SequenceBatches send sequences of batches to the ethereum
-func (etherMan *Client) SequenceBatches(ctx context.Context, sequences []ethmanTypes.Sequence, gasLimit uint64, gasPrice, nonce *big.Int) (*types.Transaction, error) {
-	if etherMan.IsReadOnly() {
-		return nil, ErrIsReadOnlyMode
-	}
-	sendSequencesOpts := *etherMan.auth
-	sendSequencesOpts.GasLimit = gasLimit
-	if gasPrice != nil {
-		sendSequencesOpts.GasPrice = gasPrice
-	} else if etherMan.GasProviders.MultiGasProvider {
-		sendSequencesOpts.GasPrice = etherMan.getGasPrice(ctx)
+// BuildSequenceBatchesTxData builds a []bytes to be sent to the PoE SC method SequenceBatches.
+func (etherMan *Client) BuildSequenceBatchesTxData(sender common.Address, sequences []ethmanTypes.Sequence) (to *common.Address, value *big.Int, data []byte, err error) {
+	opts, err := etherMan.getAuthByAddress(sender)
+	if err == ErrNotFound {
+		return nil, nil, nil, fmt.Errorf("failed to build sequence batches, err: %w", ErrPrivateKeyNotFound)
 	}
-	if nonce != nil {
-		sendSequencesOpts.Nonce = nonce
+	opts.NoSend = true
+	// force nonce, gas limit and gas price to avoid querying it from the chain
+	opts.Nonce = big.NewInt(1)
```

Comment:
This method doesn't send any tx to the chain or process this in the EVM, we just use the methods in order to build the proper Txs fields taking advantage of the code generated from the SC. 

Since we set the `NoSend` field we are safe to not get the tx sent to the chain, but we must set the fields `Nonce`, `GasLimit`, and `GasPrice` to a fixed value in order to avoid accessing the blockchain to try to estimate the gas or waste time and resources.